src/celeritas/inp/SurfacePhysics.hh

```diff
@@ -0,0 +1,206 @@
+//---------------------------------------------------------------------------//
+//! \file celeritas/inp/SurfacePhysics.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <algorithm>
+#include <map>
+
+#include "corecel/Types.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/math/SoftEqual.hh"
+#include "geocel/Types.hh"
+#include "celeritas/inp/Grid.hh"
+
+namespace celeritas
+{
+namespace inp
+{
+//---------------------------------------------------------------------------//
+// SURFACE DESCRIPTION: Reflectivity and models for surface normals.
+//---------------------------------------------------------------------------//
+/*!
+ * Surface reflectivity index, which can be a user-defined Grid, which is a
+ * function of energy (wavelength), or constant.
+ */
+struct ReflectionGrid
+{
+    Grid grid;
+
+    //! Whether the data are assigned
+    explicit operator bool() const { return static_cast<bool>(grid); }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Analytic reflectivity: use Fresnel equations.
+ */
+struct ReflectionAnalytic
+{
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Simplest surface normal treatment.
+ */
+struct Polished
+{
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Global surface normal with smearing.
+ *
+ * Roughness range is [0, 1], where 1 is maximum roughness. In Geant4 parlance,
+ * \code roughness = 1 - GetPolish(); \endcode.
+ *
+ * \note Used by the GLISUR model in Geant4.
+ */
+struct SmearRoughness
+{
+    real_type roughness{-1};  //!< [0, 1] where 1 means maximum roughness
+
+    //! Whether the data are assigned
+    explicit operator bool() const { return roughness >= 0 && roughness <= 1; }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Assumes a Gaussian distribution of the facet normal with a \f$ \sigma_\alpha
+ * \f$ standard deviation.
+ *
+ * \note Used by the Unified model in Geant4.
+ */
+struct GaussianRoughness
+{
+    real_type sigma_alpha{-1};  //!< Gaussian std. dev.
+
+    //! Whether the surface data are assigned
+    explicit operator bool() const { return sigma_alpha > 0; }
+};
+
+//---------------------------------------------------------------------------//
+//!@{
+//! \name Convenience typedef for current simplified layer implementation.
+//! \todo: Expand \c SurfaceLayer to a `map<SurfaceId, SurfaceLayerId>`, where
+//! the \c SurfaceLayerId describes a set of layers.
+using SurfaceLayer = SurfaceId;
+//!@}
+
+//---------------------------------------------------------------------------//
+// SURFACE PHYSICS: interaction mechanisms / reflection models.
+//---------------------------------------------------------------------------//
+/*!
+ * Paramaters used by different reflection mechanisms.
```

Comment:
```suggestion
 * Parameters used by different reflection mechanisms.
```
and also you can note that the three probabilities must sum to less than one at every grid point; and the remainder is diffuse scattering